src/Compilers/Core/Portable/SourceGeneration/GeneratorDriver.cs

```diff
@@ -193,7 +193,9 @@ internal GeneratorDriverState RunGeneratorsCore(Compilation compilation, Diagnos
                     {
                         generator.Initialize(pipelineContext);
                     }
-                    catch (Exception e) when (FatalError.ReportAndCatchUnlessCanceled(e, cancellationToken))
+#pragma warning disable CS0618 // ReportIfNonFatalAndCatchUnlessCanceled is obsolete
+                    catch (Exception e) when (FatalError.ReportIfNonFatalAndCatchUnlessCanceled(e, cancellationToken))
```

Comment:
@chsienki: this we observed in an earlier review -- it seems odd we're trying to report exceptions here for generator throws, when I don't think we'd do the same for other types of analyzers. (Unless @mavasani knows something I don't.)

Reply:
Agreed. We have another issue open around first-chance exceptions and cancellations, so I'll link it with that: https://github.com/dotnet/roslyn/issues/58290